src/storage-controller/src/collection_mgmt.rs

```diff
@@ -222,12 +226,25 @@ where
             }
         }
 
+        let read_only_rx =
+            if *self.read_only_rx.borrow() && migrated_storage_collections.contains(&id) {
+                // In read-only mode we create a new shard for all migrated storage collections. So we
+                // "trick" the write task into thinking that it's not in read-only mode so something is
+                // advancing this new shard.
+                assert!(id.is_system(), "unexpected non-system global id: {id:?}");
+                info!("writing to migrated differential collection {id} in read-only mode");
+                let (_, rx) = watch::channel(false);
```

Comment:
I'm somewhat worried about what the implications are of dropping the sender here. Anyone have any thoughts? My other idea was to have another instance variable that's always false and use that.

Reply:
Sounds okay, though, from the docs:
```
The value in the channel will not be dropped until the sender and all receivers have been dropped.
```